reward/src/lib.rs

```diff
@@ -0,0 +1,534 @@
+            Some(String::from("")),
+        )?;
+
+        let token: ERC20<SignerMiddleware<_, _>> = ERC20::new(signer.clone(), &address);
+        let symbol = rt.block_on(token.get_symbol())?;
+
+        term::blank();
+        term::info!(
+            "You have selected {} token",
+            term::format::highlight(term::format::bold(symbol))
+        );
+        term::blank();
+
+        decimals = rt.block_on(token.get_decimals())?.into();
+    }
+
+    let amount = options
+        .amount
+        .or_else(|| {
+            let mut amount = U256::from(0_u64);
+            while amount.is_zero() {
+                let amt: String = term::text_input("Total reward amount?", None).unwrap();
+                amount = amount_to_u256(&amt, decimals).ok()??;
+                if amount.is_zero() {
+                    term::warning("Reward amount cannot be zero");
+                }
+            }
+            Some(amount)
+        })
+        .ok_or_else(|| anyhow!("Couldn't get amount in U256"))?;
+
+    let strategy = options.strategy.unwrap_or_else(|| {
+        term::select_with_prompt(
+            "How should the reward be distributed?",
+            &["Equally", "Weighted"],
+            &"Equally",
+        )
+        .unwrap()
+        .to_string()
+    });
+
+    let rewards = calculate_rewards(strategy.to_lowercase(), amount, &all_ens)?;
+
+    // show summary and aggregate payments
+    term::blank();
+    let mut table = term::Table::default();
+    table.push([
+        term::format::bold("Address"),
+        term::format::bold("Reward"),
+        term::format::bold("ENS"),
+    ]);
+    let mut receivers = Vec::new();
+    let mut amounts = Vec::new();
+    for (ens, reward) in rewards.iter() {
+        let resolver: Result<PublicResolver<SignerMiddleware<_, _>>, _> =
+            rt.block_on(PublicResolver::get(ens, signer.clone()));
+
+        // if resolver doesn't exist, we just skip this one
+        if resolver.is_err() {
+            table.push([
+                term::format::italic(term::format::negative("Missing")),
+                term::format::tertiary(u256_to_amount(*reward, decimals)?),
+                term::format::secondary(ens),
+            ]);
+            continue;
+        }
+        let resolver = resolver.unwrap();
+
+        let address = rt
+            .block_on(resolver.address(ens))?
+            .ok_or_else(|| anyhow!("Couldn't get Address of ENS"))?;
+
+        receivers.push(address);
+        amounts.push(*reward);
+
+        table.push([
+            term::format::highlight(address),
+            term::format::tertiary(u256_to_amount(*reward, decimals)?),
+            term::format::secondary(ens),
+        ]);
+    }
+    table.render();
+    term::blank();
+
+    // exit now if this was a dry run
+    if options.dry {
+        return Ok(());
+    }
+
+    let confirm = term::select_with_prompt("Do you wish to proceed?", &["Yes", "No"], &"Yes")
+        .unwrap()
+        .to_string();
```

Comment:
Can we use a regular confirm prompt? This is a bit strange.

Reply:
I think consistency is more important right now, we can improve the confirm prompt separately if that's what we want.

Reply:
I think consistency is more important right now, we can improve the confirm prompt separately if that's what we want.

Reply:
https://github.com/radicle-dev/radicle-cli/commit/f4007e459f8ce654f3ca58ce48dc1f9910838534#diff-26d7e6053383a7793e880ae8b6ee5d2f7d7c23ccd5e88e43f44052374e9f0f58R359